src/pages/settings-page/components/ProfileEdit/hooks/use-on-data-change.ts

```diff
@@ -0,0 +1,185 @@
+  const onChangePhotoHandler = async (e: ChangeEvent<HTMLInputElement>) => {
+    const reader = new FileReader();
+    reader.onloadend = () => {
+      setFile64(reader.result);
+    };
+    const newFile = e.target.files && e.target.files[0];
+    if (newFile) {
+      setFile(newFile);
+      reader.readAsDataURL(newFile);
+    }
+  };
+  const onPhotoDelete = () => {
+    setFile(undefined);
+    setFile64(null);
+  };
+
+  const [textAreaWithDatesState, setTextAreaWithDatesState] = useState({
+    education: [
+      {
+        info: '',
+        dateStart: new Date().toISOString(),
+        dateEnd: new Date().toISOString(),
+        error: false,
+      },
+    ],
+    experience: [
+      {
+        info: '',
+        dateStart: new Date().toISOString(),
+        dateEnd: new Date().toISOString(),
+        error: false,
+      },
+    ],
+  });
+  const onChangeTextareaWithDates = (args: ITextareaWithDatesOnChange) => {
+    setTextAreaWithDatesState(prevState => {
+      const newState = { ...prevState };
+      newState[args.key][args.index] = args.item;
+      console.log(newState);
+      return newState;
+    });
+  };
+  const addField = (key: OnChangeObjectKeys) => {
+    setTextAreaWithDatesState(prevState => {
+      const newState = { ...prevState };
+      newState[key].push({
+        info: '',
+        dateStart: new Date().toISOString(),
+        dateEnd: new Date().toISOString(),
+        error: false,
+      });
+      return newState;
+    });
+  };
+
+  const [skillsOptions, setSkillsOptions] = useState([
+    { label: 'Business analysis', value: false },
+    { label: 'Consulting', value: false },
+    { label: 'Estimate', value: false },
+    { label: 'Recruiting', value: false },
+    { label: 'SMM', value: false },
+    { label: 'Copyrighting', value: false },
+    { label: 'UI/UX', value: false },
+    { label: 'Administration', value: false },
+    { label: 'Taxation', value: false },
+    { label: 'Coaching', value: false },
+    { label: 'Full stack', value: false },
+    { label: 'Quality control', value: false },
+    { label: 'Communication', value: false },
+    { label: 'JavaScript', value: false },
+    { label: 'QA Automation', value: false },
+    { label: 'ReactJS', value: false },
+    { label: 'Python', value: false },
+    { label: 'Game dev', value: false },
+    { label: 'Flutter', value: false },
+    { label: 'Node.js', value: false },
+    { label: 'DevOps', value: false },
+    { label: 'Scrum Master', value: false },
+    { label: 'Agile Coach', value: false },
+    { label: 'Project Manager', value: false },
+  ]);
+
+  const onSkillsChange = (index: number) => {
+    setSkillsOptions(prevState => {
+      return prevState.map((e, i) => {
+        if (index === i) {
+          return { ...e, value: !e.value };
+        }
+        return e;
+      });
+    });
+  };
+
+  const [englishOption, setEnglishOption] = useState('Pre-intermediate');
+  const englishOptions = [
```

Comment:
move above the hooks, dont save constants in your hooks or components

Reply:
done